scipy/stats/_multivariate.py

```diff
@@ -3775,6 +3814,52 @@ def rvs(self, eigs, random_state=None, tol=1e-13, diag_tol=1e-7):
 random_correlation = random_correlation_gen()
 
 
+class random_correlation_frozen(multi_rv_frozen):
+    def __init__(self, eigs, seed=None, tol=1e-13, diag_tol=1e-7):
+        """Create a frozen random correlation matrix distribution.
+
+        Parameters
+        ----------
+        eigs : 1d ndarray
+            Eigenvalues of correlation matrix
+        seed : {None, int, `numpy.random.Generator`,
+                `numpy.random.RandomState`}, optional
+
+            If `seed` is None (or `np.random`), the `numpy.random.RandomState`
+            singleton is used.
+            If `seed` is an int, a new ``RandomState`` instance is used,
+            seeded with `seed`.
+            If `seed` is already a ``Generator`` or ``RandomState`` instance
+            then that instance is used.
+        tol : float, optional
+            Tolerance for input parameter checks
+        diag_tol : float, optional
+            Tolerance for deviation of the diagonal of the resulting
+            matrix. Default: 1e-7
+
+        Raises
+        ------
+        RuntimeError
+            Floating point error prevented generating a valid correlation
+            matrix.
+
+        Returns
+        -------
+        rvs : ndarray or scalar
+            Random size N-dimensional matrices, dimension (size, dim, dim),
+            each having eigenvalues eigs.
+        """
+
+        self._dist = random_correlation_gen(seed)
+        self.tol = tol
+        self.diag_tol = diag_tol
+        _, self.eigs = self._dist._process_parameters(eigs, tol=self.tol)
```

Comment:
> But when we call `rvs` function of `self._dist` the `_process_parameters` function is called again,

The way univariate distributions solve the issue of calling `_process_parameters` multiple times is by monkey patching it in the `__init__` method. So, for example, here we could do:

```suggestion
        self.dims, self.eigs = self._dist._process_parameters(eigs, tol=self.tol)
        # monkey-patch _process_parameters
        def _process_parameters(eigs, tol):
            return self.dims, self.eigs
        self._dist._process_parameters = _process_parameters
```

Not we can use the public methods with `_process_parameters` being just a no-op.

Reply:
Wow! That's a straightforward way to solve this. Should we also work on monkey-patching `_process_parameters` of frozen multivariate distributions?